src/core/store/modules/app/app.actions.ts

```diff
@@ -34,11 +36,16 @@ const clearUserAppData = createAsyncThunk<void, void, ThunkAPI>('app/clearUserAp
 /*                                 Fetch Data                                 */
 /* -------------------------------------------------------------------------- */
 
-const initApp = createAsyncThunk<void, void, ThunkAPI>('app/initApp', async (_arg, { dispatch, getState }) => {
+const initApp = createAsyncThunk<void, void, ThunkAPI>('app/initApp', async (_arg, { dispatch, getState, extra }) => {
+  const { CONTRACT_ADDRESSES } = extra.config;
   const { wallet, network } = getState();
-  if (inIframe()) {
+  if (inLedgerIframe()) {
+    if (network.current !== 'mainnet') await dispatch(NetworkActions.changeNetwork({ network: 'mainnet' }));
```

Comment:
It changes the app selected network, and only triggers the prompt if its a MM wallet. Since here we know its being loaded in Ledger Live, it will not trigger the prompt.